src/NServiceBus.Core/Sagas/AttachSagaDetailsToOutGoingMessageBehavior.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Pipeline.Contexts;
+    using NServiceBus.Saga;
+    using NServiceBus.TransportDispatch;
+
+    class AttachSagaDetailsToOutGoingMessageBehavior : Behavior<OutgoingContext>
+    {
+        public override void Invoke(OutgoingContext context, Action next)
+        {
+            ActiveSagaInstance saga;
+
+            //attach the current saga details to the outgoing headers for correlation
+            if (context.TryGet(out saga) && HasBeenFound(saga))
```

Comment:
the semantics of returning a non null ActiveSagaInstance that has a `NotFound` of true confuses me